### PR TITLE
Implement nth, last, and count for iter::Copied

### PR DESCRIPTION
**Pull request by Stebalien**
Implement nth, last and count for iter::Copied.

### CODE REVIEW

**Comment by Stebalien** [comment]
(please wait on CI before reviewing)

File: src/libcore/iter/adapters/mod.rs [hunk 1]
```diff
@@ -286,6 +298,23 @@ impl<'a, I, T: 'a> Iterator for Cloned<I>
     }
 }
 
+#[stable(feature = "iter_cloned", since = "1.1.0")]
```
Comment (Stebalien) [inline src/libcore/iter/adapters/mod.rs]: does this need a new feature?

**Comment by Centril** [comment]
We currently have a moratorium on new uses of specialization: https://forge.rust-lang.org/libs/maintaining-std.html#is-specialization-involved.

**Comment by Stebalien** [comment]
Got it. I'll drop this down to just specializing `iter::Copied` (which doesn't actually use specialization).

**Comment by Stebalien** [comment]
CI is passing and this should be ready for review.

**Pull request merged**